x-pack/plugins/observability_solution/investigate/public/hooks/use_investigate_widget.tsx

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useContext, createContext } from 'react';
+import type { InvestigateWidgetCreate, WorkflowBlock } from '../../common';
+
+type UnregisterBlocksFunction = () => void;
+
+export interface UseInvestigateWidgetApi<
+  TParameters extends Record<string, any> = {},
+  TData extends Record<string, any> = {}
+> {
+  onWidgetAdd: (create: InvestigateWidgetCreate) => Promise<void>;
+  blocks: {
+    publish: (blocks: WorkflowBlock[]) => UnregisterBlocksFunction;
+  };
+}
+
+const InvestigateWidgetApiContext = createContext<UseInvestigateWidgetApi | undefined>(undefined);
+
+export const InvestigateWidgetApiContextProvider = InvestigateWidgetApiContext.Provider;
```

Comment:
@dgieselaar Is this actually used anywhere? I was looking into the inital [PR](https://github.com/elastic/kibana/pull/183293/files) of how it is being used, but I didn't find any use of it, other than being exported here. If it is not being used, then we could remove it.

Reply:
it isn't, but I did discuss it with Kevin & Bena: I think context in this case is better than prop drilling, so I would use this API instead